packages/tasks/src/model-data.ts

```diff
@@ -40,21 +40,29 @@ export interface ModelData {
 	/**
 	 * this dictionary has useful information about the model configuration
 	 */
-	config?: Record<string, unknown> & {
-		adapter_transformers?: { model_class?: string; model_name?: string };
+	config?: {
 		architectures?: string[];
+		tokenizer_config?: TokenizerConfig;
+		adapter_transformers?: {
+			model_name?: string;
+			model_class?: string;
+		};
+		diffusers?: {
+			_class_name?: string;
+		};
 		sklearn?: {
-			filename?: string;
+			model?: {
```

Comment:
Why this change? Isn't it `filename` in the Hub?

Reply:
nope not after https://github.com/huggingface/moon-landing/pull/8978 where we enforce that we always use the prop names from the actual JSON files (will make things simpler in the long run)

Reply:
Related PR https://github.com/huggingface/moon-landing/pull/9104